File: api/internal/infrastructure/persistence/payment_allocation.go
package persistence

type PaymentAllocation struct {
	PaymentID     string `gorm:"type:uuid;primaryKey"`
	GameChargeID  string `gorm:"type:uuid;primaryKey"`
	AmountInCents int    `gorm:"not null;check:amount_in_cents > 0"`

	Payment    Payment    `gorm:"foreignKey:PaymentID;constraint:OnDelete:CASCADE"`
	GameCharge GameCharge `gorm:"foreignKey:GameChargeID;constraint:OnDelete:CASCADE"`
}

File: bouncy-ui/lib/models/payment_allocation.dart
class PaymentAllocation {
  final String paymentId;
  final String gameChargeId;
  final int amountInCents;

  PaymentAllocation({
    required this.paymentId,
    required this.gameChargeId,
    required this.amountInCents,
  });

  factory PaymentAllocation.fromMap(Map<String, dynamic> map) {
    return PaymentAllocation(
      paymentId: map['PaymentID'] ?? '',
      gameChargeId: map['GameChargeID'] ?? '',
      amountInCents: map['AmountInCents'] ?? 0,
    );
  }
}
